include/dali/core/span.h

```diff
@@ -58,10 +59,28 @@ class span {
   >>
   DALI_HOST_DEV constexpr span(const span<U, Extent>& s) noexcept : data_(s.data()) {}
 
+
   DALI_HOST_DEV constexpr span(pointer firstElem, pointer lastElem) : data_(firstElem) {
     /* assert(lastElem - firstElem == Extent); */
   }
 
+  /**
+   * @remarks Container is not a specialization of span,
+   *          specialization of std::array or an array type
+   */
+  template <typename Container, typename std::enable_if<is_container<Container>::value, int>::type = 0>
```

Comment:
1. You have already `make_span` for that purpose.
2. If we have this, it should probably be `explicit`

Reply:
https://en.cppreference.com/w/cpp/container/span/span
I'm following the standard. Why do you think, this should be `explicit`?

Reply:
Without `explicit` you'll be able to do
```
std::vector<int> v = {1, 2, 3, 4, 5};
span<int> a = v;
```
which is a bit confusing IMO.

Also, I'd rather stick to the existing `make_span`
```
span<int> a = make_span(v);
```